File: python/字符串/1556. 千位分隔数.py
class Solution:
    def thousandSeparator(self, n: int) -> str:
        if n<1000:
            return str(n)
        str_n = str(n)
        i = len(str_n)-1
        res = []
        c = 0
        while i>=0:
            res.append(str_n[i])
            c+=1
            if c==3:
                res.append(".")
                c=0
            i-=1
        # print(res[::-1])
        return ''.join(list(res[::-1])).lstrip(',')
s = Solution()
print(s.thousandSeparator(12345))

File: go/string/1556. 千位分隔数.go
package string

import (
	"sort"
	"strconv"
	"strings"
)

func thousandSeparator(n int) string {
	if n < 1000 {
		return strconv.Itoa(n)
	}
	var res []string
	count := 0
	for n != 0 {
		a := n % 10
		n = n / 10
		res = append(res, strconv.Itoa(a))
		count++
		if count == 3 && n > 0 {
			res = append(res, ".")
			count = 0
		}
	}
	_ = sort.Reverse(sort.StringSlice(res))
	return strings.Join(res, "")
}
